test/jaxp/javax/xml/jaxp/unittest/transform/JDK8207760.java

```diff
@@ -101,6 +101,32 @@ public Object[][] getDataBug8207760_cdata() {
         };
     }
 
+    /*
+     * @bug 8349699
+     * Verifies that a surrogate pair at the edge of a buffer is properly handled
+     * when serializing into a Character section.
+     */
+    @Test
+    public final void testBug8349699() throws Exception {
+        String xs = "x".repeat(1017);
+        String expected = xs + "\uD835\uDF03\uD835\uDF00\uD835\uDF00\uD835\uDF00\uD835\uDF00";
```

Comment:
Would it be meaningful if we provided invalid sequences e.g, high/high and or low/low at the boundary to see if the logic successfully detect them?

Reply:
Thanks Naoto. Added test cases with invalid sequences (high/high, low/low). Also added test cases to verify both edge and normal cases.